File: leetcode/python/45. Jump Game II.py
def jump(n):
    length = len(n) - 1
    curr = -1
    ans = 0
    next = 0
    i = 0
    while next < length:
        if i > curr:
            ans += 1
            curr = next
        next = max(next, n[i] + i)
        i += 1
    return ans


print(jump([2, 3, 1, 1, 4]))

File: leetcode/kotlin/45. Jump Game II.kt
import java.io.*

internal object Main {
    var ps: PrintStream = System.out
    fun jump(n: IntArray): Int {
        val len = n.size - 1
        var curr = -1
        var next = 0
        var ans = 0
        var i = 0
        while (next < len) {
            if (i > curr) {
                ans++
                curr = next
            }
            next = Math.max(next, n[i] + i)
            i++
        }
        return ans
    }

    fun main(vararg s: String?) {
        ps.println(jump(intArrayOf(2, 3, 1, 1, 4)))
    }
}